tests/topics/test_topic_info_endpoint.py

```diff
@@ -0,0 +1,42 @@
+"""Testing TopicInfo endpoint."""
+
+import jwt
+
+from src.config import Config
+from src.topics.models import Topic
+
+
+def test_get_topic_info_endpoint_with_authorized_user_returns_correct_body(client, db_with_one_user_and_one_topic):
+    session = db_with_one_user_and_one_topic
+    topic = session.query(Topic).first()
+    authorization_token = jwt.encode({"user_id": 1}, Config.SECRET_KEY, algorithm="HS256")
+
+    result = client.get(f"/api/topics/{topic.id}", headers={"Authorization": f"Bearer {authorization_token}"})
+
+    assert result.status_code == 200
+    assert result.json == {
+        "author_id": topic.author_id,
+        "created_at": str(topic.created_at),
+        "description": topic.description,
+        "title": topic.title,
+    }
+
+
+def test_get_topic_info_endpoint_with_nonexistent_topic_returns_404_with_correct_body(client, db_with_one_user):
+    authorization_token = jwt.encode({"user_id": 1}, Config.SECRET_KEY, algorithm="HS256")
+
+    result = client.get("/api/topics/1", headers={"Authorization": f"Bearer {authorization_token}"})
+
+    assert result.status_code == 404
+    assert result.json == {"message": "Could not find a topic with id provided."}
+
+
+def test_get_topic_info_endpoint_with_invalid_token_returns_401_with_correct_body(client,
+                                                                                  db_with_one_user_and_one_topic):
+    session = db_with_one_user_and_one_topic
+    topic = session.query(Topic).first()
```

Comment:
We don't have to query topic from database here, since we already know its id we can hardcode it in the test, like this:
```python

def test_get_topic_info_endpoint_with_invalid_token_returns_401_with_correct_body(client,
                                                                                  db_with_one_user_and_one_topic):
    result = client.get(f"/api/topics/1", headers={"Authorization": "Bearer IvalidToken"})

    assert result.status_code == 401
    assert result.json == {"message": "invalid token"}
```